### PR TITLE
test: add failing test for protocol setter

### PR DESCRIPTION
**Pull request by anonrig**

### CODE REVIEW

File: include/ada/url_aggregator-inl.h [hunk 1]
```diff
@@ -429,8 +428,15 @@ inline void url_aggregator::update_base_password(const std::string_view input) {
 
   add_authority_slashes_if_needed();
 
+  // TODO: Optimization opportunity. Merge the following removal functions.
   if (input.empty()) {
     clear_base_password();
+
+    // Remove username too, if it is empty.
+    if (!has_non_empty_username()) {
+      update_base_username("");
```
Comment (lemire) [inline include/ada/url_aggregator-inl.h:437]: Why don't we have `clear_base_username()`?

In PR https://github.com/ada-url/ada/pull/308, I am proposing we add ` clear_base_hash()`.

**Review by lemire — APPROVED** [review]
I love the new tests!

File: include/ada/url_aggregator-inl.h (hunk 1 above)
Comment (anonrig) [reply include/ada/url_aggregator-inl.h:437]: I left it as a todo, because this is a really edge case. I prefer coming back to this after fixing the missing edge cases from WPT.
